lib/util.js

```diff
@@ -230,17 +230,30 @@ function format(f) {
   return str;
 }
 
-var debugs = {};
-var debugEnviron;
+const debugs = {};
+let debugEnviron;
+
+const regExpSpecialChars = /[|\\{}()[\]^$+?.]/g;
+
+function arrayToRegExp(strArr) {
+  // Escape special chars except wildcard.
+  strArr = strArr.map((str) => {
+    return str.replace(regExpSpecialChars, '\\$&')
+      .replace(/\*/g, '.*');
+  });
+
+  return new RegExp(strArr.join('|'), 'i');
```

Comment:
The RegExp must enforce start and end: `new RegExp('^' + strArr.join('|') + '$')`, otherwise any string acts as one with wildcards at the beginning and end (and since the default is an empty string, not setting NODE_DEBUG results in all debug messages being printed).

Reply:
thx, updated.